pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -710,6 +735,15 @@ func (cf *changeFrontier) noteResolvedSpan(d rowenc.EncDatum) error {
 			`unmarshalling resolved span: %x`, raw)
 	}
 
+	// TODO(ajwerner): Remove this code in 21.2.
+	if resolved.DeprecatedBoundaryReached && resolved.BoundaryType == jobspb.ResolvedSpan_NONE {
```

Comment:
I'm confused about how this conditional is ever true. Why is `DeprecatedBoundaryReached` already true if `resolved.BoundaryType` is none?

Reply:
Needs commentary. This code also is not tested :grimacing:. The idea is that nodes running the old code will only have set the boolean and not the type but the below code only wants to deal in types. I'll think about how I want to test this.